relay-metrics/src/aggregation.rs

```diff
@@ -1791,13 +1889,22 @@ impl Aggregator {
             let num_partitions = self.config.flush_partitions;
             let partitioned_buckets = self.partition_buckets(project_buckets, num_partitions);
             for (partition_key, buckets) in partitioned_buckets {
-                self.process_batches(buckets, |batch| {
+                let capped_batches =
+                    CappedBucketIter::new(buckets.into_iter(), self.config.max_flush_bytes);
+
+                let mut num_batches = 0;
+                for batch in capped_batches {
+                    relay_statsd::metric!(
+                        histogram(MetricHistograms::BucketsPerBatch) = batch.len() as f64,
+                    );
                     let batch_size = batch.len() as u64;
-                    let result = self.receiver.do_send(FlushBuckets {
+                    let receiver = self.receiver.clone();
+                    let flush = FlushBuckets {
                         project_key,
                         partition_key,
                         buckets: batch,
-                    });
+                    };
+                    let result = sender(receiver, flush).await;
```

Comment:
Note that this changes semantics of the code:

- The old version used `do_send`, which sends a fire-and-forget message.
- The new version waits for the response of `FlushBuckets`. 

Let's not change semantics of the code and use the synchronous `.send()` with a `NoResponse` on `FlushBuckets`. If that requires us to introduce a second `FlushBuckets` message elsewhere, let's rather do that. 

This will also allow you to restore `fn process_batches` above.